api-gateway-routes/api_gateway_put_function.rb

```diff
@@ -0,0 +1,11 @@
+require 'aws-sdk-s3'
+
+def lambda_handler(event:, context:)
+  region = context.invoked_function_arn.split(':')[3]
```

Comment:
Not critical, but there's an [ARN class](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/ARN.html) with handy methods for parsing the ARN and accessing the different logical identifiers within it.